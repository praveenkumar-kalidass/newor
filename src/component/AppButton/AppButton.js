import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'native-base';

import useTheme from '../../theme/useTheme';

const AppButton = ({ variant, ...other }) => {
  const theme = useTheme();
  const bg = useMemo(() => theme.color[`${variant.toUpperCase()}_100`], [variant]);
  const pressedBG = useMemo(() => theme.color[`${variant.toUpperCase()}_20`], [variant]);

  return (
    <Button
      bg={bg}
      _pressed={{ bg: pressedBG }}
      _text={{ fontWeight: 'bold' }}
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...other}
    />
  );
};

AppButton.propTypes = {
  variant: PropTypes.oneOf(['primary', 'secondary']).isRequired,
};

export default AppButton;
