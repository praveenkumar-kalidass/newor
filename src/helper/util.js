import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
