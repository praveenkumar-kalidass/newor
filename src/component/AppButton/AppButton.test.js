import React from 'react';

import { render } from 'test/util';
import AppButton from './AppButton';

describe('AppButton', () => {
  it('should render primary button', () => {
    expect(render(<AppButton variant="primary" />)).toMatchSnapshot();
  });

  it('should render secondary button', () => {
    expect(render(<AppButton variant="secondary" />)).toMatchSnapshot();
  });
});
