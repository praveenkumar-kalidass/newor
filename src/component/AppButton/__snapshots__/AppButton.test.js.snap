// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton should render primary button 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ad0000",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 10,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      test={true}
    />
  </View>
</RNCSafeAreaProvider>
`;

exports[`AppButton should render secondary button 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00238a",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 10,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      test={true}
    />
  </View>
</RNCSafeAreaProvider>
`;
